src/chains/arbitrum/vm/opcodes/block/number.ts

```diff
@@ -0,0 +1,31 @@
+import { Opcode } from "@/chains/types";
+
+export const number: Opcode = {
+  number: 43,
+  name: 'number',
+  description: 'L2 block number',
```

Comment:
This is the only difference between the mainnet `number` opcode and the arbitrum one (as well as a reference link too).

Reply:
Based on the reference I think this actually returns the L1 block number, and you have to call the `arbBlockNumber` on the ArbSys contract to get Arbitrum's own L2 block number?

Reply:
I think you're right, I got it wrong

Reply:
"As a general rule, any timing assumptions a contract makes about block numbers and timestamps should be considered generally reliable in the longer term (i.e., on the order of at least several hours) **but unreliable in the shorter term (minutes)**." ([source](https://developer.arbitrum.io/time#ethereum-block-numbers-within-arbitrum)) - I feel that it's something we should underline though if we want to be thorough.

Reply:
I guess I was too lazy to try... 🫠
Let me add the difference back again